Explorer/Assets/DCL/LOD/Components/SceneLODInfo.cs

```diff
@@ -39,26 +38,25 @@ public void Dispose(World world)
 
         public void SetCurrentLOD(LODAsset newLod)
         {
-            if (newLod.LoadingFailed)
-                ProfilingCounters.Failling_LOD_Amount.Value++;
-            else
+            CurrentLOD = newLod;
+            UpdateLastSuccessfullLOD();
+        }
+
+        public void UpdateLastSuccessfullLOD()
+        {
+            if (CurrentLOD.State == LODAsset.LOD_STATE.SUCCESS)
             {
-                if (!newLod.LodKey.Equals(LastSuccessfulLOD))
-                {
-                    LastSuccessfulLOD?.Release();
-                    LastSuccessfulLOD = newLod;
-                }
+                LastSuccessfulLOD?.Release();
+                LastSuccessfulLOD = CurrentLOD;
             }
-
-            CurrentLOD = newLod;
         }
 
-        public LODAsset? GetCurrentLOD()
+        public LODAsset GetCurrentLOD()
         {
             return CurrentLOD;
         }
 
-        public LODAsset? GetCurrentSuccessfulLOD()
+        public LODAsset GetCurrentSuccessfulLOD()
         {
             return LastSuccessfulLOD;
         }
```

Comment:
`ResetToCurrentSuccesfullLOD` might be not correct: there is a significant difference when you switched (?) to a class: you equalize their reference instead of the content. I am not sure what was your goal. Please review